addons/pos_loyalty/static/src/overrides/components/control_buttons/control_buttons.js

```diff
@@ -52,11 +52,16 @@ patch(ControlButtons.prototype, {
                 });
             }
             if (selectedProgram) {
-                this.pos.addProductFromUi(selectedProgram.trigger_product_ids[0], {
-                    price: -orderTotal,
-                    merge: false,
-                    eWalletGiftCardProgram: selectedProgram,
-                });
+                this.pos.addLineToCurrentOrder(
+                    {
+                        product_id: selectedProgram.trigger_product_ids[0],
```

Comment:
Price unit should be in the first object as value object is used to create the orderline in `addLineToCurrentOrder`

```
        const values = {
            price_extra: 0,
            price_unit: 0,
            order_id: this.get_order(),
            qty: 1,
            tax_ids: product.taxes_id[0] ? [["link", product.taxes_id[0]]] : [],
            ...vals,
        };
```